src/cwe_checker_lib/src/checkers/cwe_789.rs

```diff
@@ -0,0 +1,176 @@
+fn multiply_args_for_calloc(
+    pir: &PointerInference,
+    jmp_tid: &Tid,
+    parms: Vec<&Arg>,
+) -> Option<DataDomain<IntervalDomain>> {
+    if let (Some(nmeb), Some(size)) = (
+        pir.eval_parameter_arg_at_call(jmp_tid, parms[0]),
+        pir.eval_parameter_arg_at_call(jmp_tid, parms[1]),
+    ) {
+        nmeb.bin_op(BinOpType::IntMult, &size);
+        return Some(nmeb);
+    }
+    None
+}
+
+/// Generate the CWE warning for a detected instance of the CWE.
+fn generate_cwe_warning(allocation: &Tid, is_stack_allocation: bool) -> CweWarning {
+    CweWarning::new(
+        CWE_MODULE.name,
+        CWE_MODULE.version,
+        format!(
+            "(Large memory allocation) Potential {}memory exhaustion at {}",
+            match is_stack_allocation {
+                true => "stack ",
+                false => "",
+            },
+            allocation.address
+        ),
+    )
+    .tids(vec![format!("{}", allocation)])
+    .addresses(vec![allocation.address.clone()])
+    .symbols(vec![])
+}
+
+/// Run the CWE check.
+/// For each function, we check calls of the defined functions and instructions that
+/// assign a value to the stackpointer.
+pub fn check_cwe(
+    analysis_results: &AnalysisResults,
+    cwe_params: &serde_json::Value,
+) -> (Vec<LogMessage>, Vec<CweWarning>) {
+    let project = analysis_results.project;
+    let config: Config = serde_json::from_value(cwe_params.clone()).unwrap();
+    let mut cwe_warnings = Vec::new();
+    let pir = analysis_results.pointer_inference.unwrap();
+    let symbol_map = get_symbol_map(project, &config.symbols);
+
+    for sub in project.program.term.subs.values() {
+        // Function call allocation case
+        for (_, jump, symbol) in get_callsites(sub, &symbol_map) {
+            if let Some(interval) = match symbol.name.as_str() {
+                "calloc" => multiply_args_for_calloc(
+                    pir,
+                    &jump.tid,
+                    vec![&symbol.parameters[0], &symbol.parameters[1]],
+                ),
+                "realloc" => pir.eval_parameter_arg_at_call(&jump.tid, &symbol.parameters[1]),
+                _ => pir.eval_parameter_arg_at_call(&jump.tid, &symbol.parameters[0]),
+            } {
+                if exceeds_threshold_on_call(interval, config.heap_threshold) {
+                    cwe_warnings.push(generate_cwe_warning(&jump.tid, false));
+                }
+            }
+        }
+        // Stack allocation case
+        for blk in &sub.term.blocks {
+            let assign_on_sp: Vec<&Term<Def>> = blk
+                .term
+                .defs
+                .iter()
+                .filter(|x| is_assign_on_sp(&x.term, &project.stack_pointer_register))
+                .collect();
+            for assign in assign_on_sp {
+                if let Some(interval) = analysis_results
+                    .pointer_inference
+                    .unwrap()
```

Comment:
You have already stored the result of `analysis_results.pointer_inference.unwrap()` in the `pir` variable.